### DISCUSSION TITLE
Make RenderController Index action async

### DISCUSSION THREAD

**Discussion by umbracotrd** [umbraco/Umbraco-CMS#17562] [Features and ideas]
The `RenderController.cs` > `Index` action is synchronous which prevents the derived classes call async methods inside `Index` method.

There was [a discussion in the forum](https://our.umbraco.com/forum/umbraco-9/106051-async-index-method-in-render-controller) a while ago and there were few hacky ways to do it but this era of asynchronous programming, I think it would be nice that umbraco cms supports this out of the box.

**Comment by bjarnef** [comment]
It is mentioned in docs here how to used `async` action.
https://docs.umbraco.com/umbraco-cms/reference/routing/custom-controllers#example-hijacking-route-requests-to-a-product-page

You would need to hide the default inherited action with `[NonAction]`:

```
public class ProductPageController : RenderController
{
    [NonAction]
    public sealed override IActionResult Index() => throw new NotImplementedException();
    public async Task<IActionResult> Index(CancellationToken cancellationToken)
    {
        await SomethingAsync(cancellationToken);
        return CurrentTemplate(CurrentPage);
    }
}
```

**Comment by umbracotrd** [comment]
Thanks @bjarnef , I should've rtfm 😅
